cmd/root.go

```diff
@@ -18,396 +18,436 @@ package cmd
 
 import (
 	"bufio"
-	"errors"
-	"flag"
 	"fmt"
+	"github.com/sonatype-nexus-community/nancy/configuration"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
 	"github.com/common-nighthawk/go-figure"
 	"github.com/golang/dep"
 	"github.com/mitchellh/go-homedir"
 	"github.com/sirupsen/logrus"
+	"github.com/sonatype-nexus-community/go-sona-types/ossindex"
+	ossIndexTypes "github.com/sonatype-nexus-community/go-sona-types/ossindex/types"
 	"github.com/sonatype-nexus-community/nancy/audit"
 	"github.com/sonatype-nexus-community/nancy/buildversion"
-	"github.com/sonatype-nexus-community/nancy/configuration"
 	"github.com/sonatype-nexus-community/nancy/customerrors"
-	. "github.com/sonatype-nexus-community/nancy/logger"
-	"github.com/sonatype-nexus-community/nancy/ossindex"
+	"github.com/sonatype-nexus-community/nancy/logger"
 	"github.com/sonatype-nexus-community/nancy/packages"
 	"github.com/sonatype-nexus-community/nancy/parse"
 	"github.com/sonatype-nexus-community/nancy/types"
 	"github.com/spf13/cobra"
-	"os"
-	"path/filepath"
-	"reflect"
-	"strings"
 
 	"github.com/spf13/viper"
 )
 
-var cfgFile string
+type ossiServerFactory interface {
+	create() ossindex.IServer
+}
 
-var configOssi configuration.Configuration
-var excludeVulnerabilityFilePath string
-var outputFormat string
+type ossiFactory struct{}
+
+func (ossiFactory) create() ossindex.IServer {
+	server := ossindex.New(logLady, ossIndexTypes.Options{
+		Username:    viper.GetString(configuration.YamlKeyUsername),
+		Token:       viper.GetString(configuration.YamlKeyToken),
+		Tool:        "nancy-client",
+		Version:     buildversion.BuildVersion,
+		DBCacheName: "nancy-cache",
+		TTL:         time.Now().Local().Add(time.Hour * 12),
+	})
+
+	if logLady != nil && logLady.IsLevelEnabled(logrus.DebugLevel) {
```

Comment:
Ditto same question @DarthHater asked on the iq side of things

Reply:
> Ditto same question @DarthHater asked on the iq side of things

Oof! Good catch. I think this was leftover thrashing when some unit tests had not yet initialized the null logger. 
Fixed: 4f842ef